security/pfSense-pkg-suricata/files/usr/local/www/suricata/suricata_global.php

```diff
@@ -134,90 +135,90 @@
+		config_set_path('installedpackages/suricata/config/0/extra_rules', $extra_rules);
 
 		// If any rule sets are being turned off, then remove them
 		// from the active rules section of each interface.  Start
 		// by building an arry of prefixes for the disabled rules.
 		$disabled_rules = array();
 		$disable_ips_policy = false;
-		if ($config['installedpackages']['suricata']['config'][0]['enable_vrt_rules'] == 'off') {
+		if (config_get_path('installedpackages/suricata/config/0/enable_vrt_rules') == 'off') {
 			$disabled_rules[] = VRT_FILE_PREFIX;
 			$disable_ips_policy = true;
 		}
-		if ($config['installedpackages']['suricata']['config'][0]['snortcommunityrules'] == 'off')
+		if (config_get_path('installedpackages/suricata/config/0/snortcommunityrules') == 'off')
 			$disabled_rules[] = GPL_FILE_PREFIX;
-		if ($config['installedpackages']['suricata']['config'][0]['enable_etopen_rules'] == 'off')
+		if (config_get_path('installedpackages/suricata/config/0/enable_etopen_rules') == 'off')
 			$disabled_rules[] = ET_OPEN_FILE_PREFIX;
-		if ($config['installedpackages']['suricata']['config'][0]['enable_etpro_rules'] == 'off')
+		if (config_get_path('installedpackages/suricata/config/0/enable_etpro_rules') == 'off')
 			$disabled_rules[] = ET_PRO_FILE_PREFIX;
 
-		if ($config['installedpackages']['suricata']['config'][0]['enable_feodo_botnet_c2_rules'] == 'off')
+		if (config_get_path('installedpackages/suricata/config/0/enable_feodo_botnet_c2_rules') == 'off')
 			$disabled_rules[] = "feodotracker";
-		if ($config['installedpackages']['suricata']['config'][0]['enable_abuse_ssl_blacklist_rules'] == 'off')
+		if (config_get_path('installedpackages/suricata/config/0/enable_abuse_ssl_blacklist_rules') == 'off')
 			$disabled_rules[] = "sslblacklist_tls_cert";
 
 		if (empty($enabled_extra_rules))
 			$disabled_rules[] = EXTRARULE_FILE_PREFIX;
 
 		// Now walk all the configured interface rulesets and remove
 		// any matching the disabled ruleset prefixes.
-		if (is_array($config['installedpackages']['suricata']['rule'])) {
-			foreach ($config['installedpackages']['suricata']['rule'] as &$iface) {
-				// Disable Snort IPS policy if Snort rules are disabled
-				if ($disable_ips_policy) {
-					$iface['ips_policy_enable'] = 'off';
-					unset($iface['ips_policy']);
-				}
-				$enabled_rules = explode("||", $iface['rulesets']);
-				foreach ($enabled_rules as $k => $v) {
-					foreach ($disabled_rules as $d) {
-						if (strpos(trim($v), $d) !== false) { 
-							unset($enabled_rules[$k]);
-							continue;
-						} elseif (!empty($enabled_extra_rules)) {
-							foreach ($enabled_extra_rules as $exrule) {
-								if (strpos(trim($v), EXTRARULE_FILE_PREFIX . $exrule)) {
-									unset($enabled_rules[$k]);
-									continue 2;
-								}
+		$a_ifaces = config_get_path('installedpackages/suricata/rule', []);
+		foreach ($a_ifaces as $iface) {
```

Comment:
Should be `&$iface`

Reply:
Should have been an array reference. Fixed it.